nameable.rb

```diff
@@ -0,0 +1,34 @@
+# Class Nameable
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+# Class Decorator inhereting from Nameable
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# Class CapitalizeDecorator inhereting from Decorator
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.split.map(&:capitalize).join(' ')
+  end
+end
+
+# Class TrimmerDecorator inhereting from Decorator
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.strip
```

Comment:
- Hey, using the strip method here won't check the string to ensure the character is not more than 10. You can read more about strip [here](https://www.educative.io/answers/what-is-the-strip-method-in-ruby), instead, you can use another method. Here's an example:

` @nameable.correct_name.split.first(10).join(' ')` or 
`@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`